packages/main/src/plugin/container-registry.ts

```diff
@@ -1265,6 +1265,66 @@ export class ContainerProviderRegistry {
     return crypto.createHash('sha512').update(imageName).digest('hex');
   }
 
+  async updateImage(engineId: string, imageId: string): Promise<void> {
+    let telemetryOptions = {};
+    try {
+      // first get image info to check if it has a registry
+      const imageInfo = await this.getMatchingImage(engineId, imageId).inspect();
+      if (!imageInfo.RepoTags || imageInfo.RepoTags.length === 0) {
+        throw new Error('Image has no registry tags and cannot be updated');
+      }
+      // get the first repo tag (we know it exists because length > 0)
+      const repoTag = imageInfo.RepoTags[0]!;
```

Comment:
suggestion: get the repoTag without the `!`, and test if is defined (as in line 1273).